parquet-cli/src/main/java/org/apache/parquet/cli/commands/ColumnSizeCommand.java

```diff
@@ -0,0 +1,132 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.cli.commands;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.avro.file.SeekableInput;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.cli.BaseCommand;
+import org.apache.parquet.cli.util.Formats;
+import org.apache.parquet.format.converter.ParquetMetadataConverter;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Parameters(commandDescription="Print the column sizes of a parquet file")
+public class ColumnSizeCommand extends BaseCommand {
+
+  public ColumnSizeCommand(Logger console) {
+    super(console);
+  }
+
+  @Parameter(description = "<parquet path>")
+  String target;
+
+  @Parameter(
+    names = {"-c", "--column", "--columns"},
+    description = "List of columns",
+    required = false)
+  List<String> columns;
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public int run() throws IOException {
+    Preconditions.checkArgument(target != null,
+      "A Parquet file is required.");
+
+    Path inputFile = new Path(target);
+    Map<String, Long> columnSizes = getColumnSizeInBytes(inputFile);
+    Map<String, Float> columnPercentage = getColumnPercentage(columnSizes);
+
+    // If user inputted columns, only print out size for those columns
+    if (columns != null && columns.size() > 0) {
+      for (String inputColumn : columns) {
+        long size = 0;
+        float percentage = 0;
+        for (String column : columnSizes.keySet()) {
+          if (column.startsWith(inputColumn)) {
+            size += columnSizes.get(column);
+            percentage += columnPercentage.get(column);
+          }
+        }
+        console.info(inputColumn + "->" + " Size In Bytes: " + size + " Size In Percentage: " + percentage);
+      }
+    } else {
+      for (String column : columnSizes.keySet()) {
+        console.info(column + "->" + " Size In Bytes: " + columnSizes.get(column)
+          + " Size In Percentage: " + columnPercentage.get(column));
+      }
+    }
+
+    return 0;
+  }
+
+  @Override
+  public List<String> getExamples() {
+    return Lists.newArrayList(
+        "# Print every column size in byte and percentage for a Parquet file",
+        "sample.parquet",
+        "sample.parquet col_1 col_2",
+        "sample.parquet col_1 col_2.sub_col_a"
```

Comment:
I think, one of `-c`, `--column` or `--columns` is missing from here

Reply:
Got it.